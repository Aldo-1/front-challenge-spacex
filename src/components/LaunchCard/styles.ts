import styled, { css } from 'styled-components'

const baseTitleStyles = css`
  color: #273944;
  margin-bottom: 1.2rem;
`

export const Wrapper = styled.a`
  background-color: white;
  text-decoration: none;
  display: flex;
  flex-direction: column;
  border: 2px solid #c2d0d8;
  padding: 14px 10px;
`

export const Title = styled.h4`
  ${baseTitleStyles}
  font-size: 1.8rem;
`

export const Description = styled.p`
  ${baseTitleStyles}
  font-size: 1.6rem;
`

export const Date = styled.time`
  ${baseTitleStyles}
  margin-bottom: 0px;
  font-size: 1.6rem;
  font-style: italic;
  align-self: flex-end;
`
