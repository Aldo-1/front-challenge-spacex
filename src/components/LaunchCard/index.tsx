import { formatDate } from 'utils/formatDate'
import * as S from './styles'

export type LaunchCardProps = {
  mission_name: string
  mission_link: string
  launch_date_local: string
}

const LaunchCard = ({
  mission_name,
  mission_link,
  launch_date_local
}: LaunchCardProps) => (
  <S.Wrapper href={`/mission/${mission_link}`}>
    <S.Title>{mission_name}</S.Title>
    <S.Description>
      This {mission_name} will launch the sixth batch of operational Starlink
      satellites, which are expected to be version 1.0, from SLC-40
    </S.Description>
    <S.Date>{formatDate(launch_date_local)}</S.Date>
  </S.Wrapper>
)

export default LaunchCard
