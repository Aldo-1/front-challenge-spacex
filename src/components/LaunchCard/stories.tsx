import { Story, Meta } from '@storybook/react/types-6-0'
import LaunchCard, { LaunchCardProps } from '.'

export default {
  title: 'LaunchCard',
  component: LaunchCard
} as Meta

export const Default: Story<LaunchCardProps> = (args) => (
  <LaunchCard {...args} />
)

Default.args = {
  mission_name: 'Starlink-15 (v1.0)',
  mission_link: 'https://www.google.com.br',
  launch_date_local: '2020-10-24T11:31:00-04:00'
}
