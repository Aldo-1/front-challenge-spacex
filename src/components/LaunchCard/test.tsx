import { render, screen } from '@testing-library/react'

import LaunchCard from '.'

const props = {
  mission_name: 'Starlink-15',
  mission_link: 'https://www.google.com.br',
  launch_date: '2020-10-24T11:31:00-04:00'
}

describe('<LaunchCard />', () => {
  it('should render the heading', () => {
    const { container } = render(<LaunchCard {...props} />)

    expect(
      screen.getByRole('heading', { name: /Starlink-15/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the description', () => {
    const { container } = render(<LaunchCard {...props} />)

    expect(
      screen.getByText(
        `This ${props.mission_name} will launch the sixth batch of operational Starlink satellites, which are expected to be version 1.0, from SLC-40`
      )
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the correct date', () => {
    const { container } = render(<LaunchCard {...props} />)

    expect(screen.getByText('24/10/2020')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
