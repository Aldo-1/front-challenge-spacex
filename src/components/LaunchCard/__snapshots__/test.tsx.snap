// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LaunchCard /> should render the correct date 1`] = `
.c0 {
  background-color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 2px solid #c2d0d8;
  padding: 14px 10px;
}

.c1 {
  color: #273944;
  margin-bottom: 1.2rem;
  font-size: 1.8rem;
}

.c2 {
  color: #273944;
  margin-bottom: 1.2rem;
  font-size: 1.6rem;
}

.c3 {
  color: #273944;
  margin-bottom: 1.2rem;
  margin-bottom: 0px;
  font-size: 1.6rem;
  font-style: italic;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

<a
  class="c0"
  href="https://www.google.com.br"
  target="_blank"
>
  <h4
    class="c1"
  >
    Starlink-15
  </h4>
  <p
    class="c2"
  >
    This 
    Starlink-15
     will launch the sixth batch of operational Starlink satellites, which are expected to be version 1.0, from SLC-40
  </p>
  <time
    class="c3"
  >
    24/10/2020
  </time>
</a>
`

exports[`<LaunchCard /> should render the description 1`] = `
.c0 {
  background-color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 2px solid #c2d0d8;
  padding: 14px 10px;
}

.c1 {
  color: #273944;
  margin-bottom: 1.2rem;
  font-size: 1.8rem;
}

.c2 {
  color: #273944;
  margin-bottom: 1.2rem;
  font-size: 1.6rem;
}

.c3 {
  color: #273944;
  margin-bottom: 1.2rem;
  margin-bottom: 0px;
  font-size: 1.6rem;
  font-style: italic;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

<a
  class="c0"
  href="https://www.google.com.br"
  target="_blank"
>
  <h4
    class="c1"
  >
    Starlink-15
  </h4>
  <p
    class="c2"
  >
    This 
    Starlink-15
     will launch the sixth batch of operational Starlink satellites, which are expected to be version 1.0, from SLC-40
  </p>
  <time
    class="c3"
  >
    24/10/2020
  </time>
</a>
`

exports[`<LaunchCard /> should render the heading 1`] = `
.c0 {
  background-color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 2px solid #c2d0d8;
  padding: 14px 10px;
}

.c1 {
  color: #273944;
  margin-bottom: 1.2rem;
  font-size: 1.8rem;
}

.c2 {
  color: #273944;
  margin-bottom: 1.2rem;
  font-size: 1.6rem;
}

.c3 {
  color: #273944;
  margin-bottom: 1.2rem;
  margin-bottom: 0px;
  font-size: 1.6rem;
  font-style: italic;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

<a
  class="c0"
  href="https://www.google.com.br"
  target="_blank"
>
  <h4
    class="c1"
  >
    Starlink-15
  </h4>
  <p
    class="c2"
  >
    This 
    Starlink-15
     will launch the sixth batch of operational Starlink satellites, which are expected to be version 1.0, from SLC-40
  </p>
  <time
    class="c3"
  >
    24/10/2020
  </time>
</a>
`
